Rlabkey/R/labkey.webdav.R

```diff
@@ -0,0 +1,230 @@
+    } else {
+        response <- GET(url=url, write_disk(localFilePath, overwrite=overwrite), config=options)
+    }
+
+    processResponse(response)
+}
+
+labkey.webdav.put <- function(localFile, baseUrl=NULL, folderPath, remoteFilePath, fileSet="@files")
+{
+    if (missing(localFile)) {
+        stop (paste("A value must be specified for localFile"))
+    }
+
+    if (!file.exists(localFile)){
+        stop (paste0("File does not exist: ", localFile));
+    }
+
+    url <- labkey.webdav._validateRemoteUrl(baseUrl = baseUrl, folderPath = folderPath, fileSet = fileSet, remoteFilePath = remoteFilePath)
+    options <- labkey.getRequestOptions(method="POST")
+
+    pbody <- upload_file(localFile)
+
+    if (!is.null(.lkdefaults[["debug"]]) && .lkdefaults[["debug"]] == TRUE) {
+        print(paste0("URL: ", url))
+        response <- PUT(url=url, config=options, body=pbody, verbose(data_in=TRUE, info=TRUE, ssl=TRUE))
+    } else {
+        response <- PUT(url=url, config=options, body=pbody)
+    }
+
+    processResponse(response, responseType = "text/plain; charset=utf-8")
+
+    return(TRUE)
+}
+
+labkey.webdav.mkDir <- function(baseUrl=NULL, folderPath, remoteFilePath, fileSet="@files")
+{
+    url <- labkey.webdav._validateRemoteUrl(baseUrl = baseUrl, folderPath = folderPath, fileSet = fileSet, remoteFilePath = remoteFilePath)
+
+    options <- labkey.getRequestOptions(method="POST")
+
+    if (!is.null(.lkdefaults[["debug"]]) && .lkdefaults[["debug"]] == TRUE) {
+        print(paste0("URL: ", url))
+        response <- VERB("MKCOL", url=url, config=options, verbose(data_in=TRUE, info=TRUE, ssl=TRUE))
+    } else {
+        response <- VERB("MKCOL", url=url, config=options)
+    }
+
+    processResponse(response, responseType = "text/plain; charset=utf-8")
+
+    return(TRUE)
+}
+
+labkey.webdav._validateRemoteUrl <- function(baseUrl=NULL, folderPath, remoteFilePath, fileSet="@files")
+{
+    baseUrl=labkey.getBaseUrl(baseUrl);
+
+    ## check required parameters
+    if (missing(baseUrl) || is.null(baseUrl) || missing(folderPath) || missing(fileSet) || missing(remoteFilePath)){
+        stop (paste("A value must be specified for each of baseUrl, folderPath, fileSet, and remoteFilePath"));
+    }
+    
+    ## normalize the folder path
+    folderPath <- encodeFolderPath(folderPath);
+
+    return(paste(baseUrl, "_webdav", folderPath, fileSet, "/", remoteFilePath, sep=""))
+}
+
+labkey.webdav.pathExists <- function(baseUrl=NULL, folderPath, remoteFilePath, fileSet="@files")
+{
+    baseUrl=labkey.getBaseUrl(baseUrl);
+
+    if (missing(baseUrl) || is.null(baseUrl) || missing(folderPath) || missing(remoteFilePath)) {
+        stop (paste("A value must be specified for each of baseUrl, folderPath, fileSet, and remoteFilePath"));
+    }
+    
+    ret <- labkey.webdav.listDir(baseUrl, folderPath, fileSet, remoteFilePath)
```

Comment:
This probably works for R, but it's confusing to non R experts that fileSet and remoteFilePath are swapped in order

Reply:
well that's what I originally had, but @labkey-klum suggested the re-order, given it has a default value.  I tend to agree w/ Karl; however, i dont love how R deals with arguments in general.